src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2826,33 +2826,29 @@ protected function asDateTime($value)
         // these checks since they will be a waste of time, and hinder performance
         // when checking the field. We will just return the DateTime right away.
         if ($value instanceof DateTime) {
```

Comment:
It would call uselessly `Carbon::instance`, `Carbon construct` and `Datetime construct` for an object that we know is already a `Carbon`